src/main/java/org/apache/lucene/analysis/ComboAnalyzer.java

```diff
@@ -168,113 +157,109 @@ public ComboAnalyzer disableDeduplication() {
         return this;
     }
 
-    protected ReaderCloneFactory.ReaderCloner<? extends Reader> cloneReader(Reader originalReader) {
-        ReaderCloneFactory.ReaderCloner<? extends Reader> rtn;
+    private static Tokenizer DUMMY_TOKENIZER = new Tokenizer(){
+        @Override
+        public boolean incrementToken() throws IOException {
+            return false;
+        }
+    };
 
-        // Duplication of the original reader, to feed all sub-analyzers
-        if (subAnalyzers.length <= 1) {
+    @Override
+    protected TokenStreamComponents createComponents(String fieldName) {
+        return new CombiningTokenStreamComponents(fieldName);
```

Comment:
So, this is the largest change in code. In Lucene 5, components are more or less forced to be reusable. 

Instead of having an external TokenStreamComponents class, there is now an internal one. The internal tokenstreams class will generate the tokenStreams when the tokenStreams() method is called, not before that. In my opinion, this makes this file a bit more readable, since the state is all embedded in the subclass class instead. 

The basic reuse/deduplication is still the same, but methods are smaller. Would love feedback.